libvcx/src/api/vcx.rs

```diff
@@ -145,6 +145,31 @@ pub extern fn vcx_open_wallet(command_handle: CommandHandle, cb: extern fn(xcomm
     error::SUCCESS.code_num
 }
 
+/// Creates agent client, which is necessary for communication with the agency. Can be called only after wallet has been opened via vcx_open_wallet.
+///
+/// #Params
+/// command_handle: command handle to map callback to user context.
+/// wallet_handle: wallet handle for the wallet to be used by the agent client (currently not used)
+/// config: config of the agent client (currently not used)
+///
+/// cb: Callback that provides error status of initialization
+///
+/// #Returns
+/// Error code as a u32
+#[no_mangle]
+pub extern fn vcx_init_agency_client(command_handle: CommandHandle, wallet_handle: WalletHandle, config: *const c_char, cb: extern fn(xcommand_handle: CommandHandle, err: u32)) -> u32 {
```

Comment:
Let's hold off from extending API just yet. Let's keep this PR internal changes and refactoring only, then create issue with new API proposal (onboarding new agent, initializing existing agent, creating wallet, opening wallet, etc. ) - then implement those APIs in separate PR.

Reply:
Or maybe before thinking about `aries-vcx` apis, we can first define `agency_client` APIs for creation and initialization:
- create new client (create it in agency)
- create client instance (create client/initialize create which can be used to talk to agency)

#### Create new client
I think all we need to create new client is: `agency_url`, `agency_did`, `agency_verkey` and a wallet handle. 
The output of this operation should be whatever is need to create client instance. I'd suggest to output to be JSON config, just as it is currently on `vcx_provision_agent`

#### Create client instance / initialization
The required information here should be
- agency_url
- agency_did
- agency_verkey
- remote_to_sdk_did
- remote_to_sdk_verkey
- sdk_to_remote_did
- sdk_to_remote_verkey
and wallet handle, whereas it's up to `agency_client` consumer responsibility to pass in the correct wallet handle.

Reply:
However this should not yet impact `aries-vcx`. Let's first design good api for the `agency_client`, then we can enhance API for `aries-vcx`